src/sender.js

```diff
@@ -542,39 +897,75 @@ function timestampToNanos(timestamp, unit) {
     }
 }
 
-function constructJwk(options) {
-    if (options) {
-        if (options.auth) {
-            if (!options.auth.keyId) {
-                throw new Error('Missing username, please, specify the \'keyId\' property of the \'auth\' config option. ' +
-                    'For example: new Sender({auth: {keyId: \'username\', token: \'private key\'}})');
-            }
-            if (typeof options.auth.keyId !== 'string') {
-                throw new Error('Please, specify the \'keyId\' property of the \'auth\' config option as a string. ' +
-                    'For example: new Sender({auth: {keyId: \'username\', token: \'private key\'}})');
-            }
-            if (!options.auth.token) {
-                throw new Error('Missing private key, please, specify the \'token\' property of the \'auth\' config option. ' +
-                    'For example: new Sender({auth: {keyId: \'username\', token: \'private key\'}})');
-            }
-            if (typeof options.auth.token !== 'string') {
-                throw new Error('Please, specify the \'token\' property of the \'auth\' config option as a string. ' +
-                    'For example: new Sender({auth: {keyId: \'username\', token: \'private key\'}})');
-            }
+function initSenderOptions(options) {
+    if (!options) {
+        options = {};
+    }
 
-            return {
-                kid: options.auth.keyId,
-                d: options.auth.token,
-                ...PUBLIC_KEY,
-                kty: 'EC',
-                crv: 'P-256'
-            };
-        } else {
-            return options.jwk;
+    // defaults to TCP for backwards compatibility
```

Comment:
AFAIR we broke backward compatibility in other clients: we ask the user to specify the protocol to use now. @jerrinot please correct me if I'm wrong.

Reply:
made the protocol option mandatory